cmake/BLTMacros.cmake

```diff
@@ -760,27 +760,43 @@ macro(blt_append_custom_compiler_flag)
    if(NOT DEFINED arg_FLAGS_VAR)
       message( FATAL_ERROR "append_custom_compiler_flag macro requires FLAGS_VAR keyword and argument." )
    endif()
-
+   
    # Set the desired flags based on the compiler family   
-   if( DEFINED arg_CLANG AND COMPILER_FAMILY_IS_CLANG )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_CLANG} " )
-   elseif( DEFINED arg_XL AND COMPILER_FAMILY_IS_XL )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_XL} " )
-   elseif( DEFINED arg_INTEL AND COMPILER_FAMILY_IS_INTEL )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_INTEL} " )
-   elseif( DEFINED arg_GNU AND COMPILER_FAMILY_IS_GNU )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_GNU} " )
-   elseif( DEFINED arg_HCC AND COMPILER_FAMILY_IS_HCC )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_HCC} " )
-    elseif( DEFINED arg_PGI AND COMPILER_FAMILY_IS_PGI )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_PGI} " )
-   elseif( DEFINED arg_MSVC_INTEL AND COMPILER_FAMILY_IS_MSVC_INTEL )
+   # MSVC COMPILER FAMILY applies to C/C++ and Fortran
+   string(TOLOWER ${arg_FLAGS_VAR} lower_flag_var)
+   if( DEFINED arg_MSVC_INTEL AND COMPILER_FAMILY_IS_MSVC_INTEL )
       set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_MSVC_INTEL} " )
    elseif( DEFINED arg_MSVC AND (COMPILER_FAMILY_IS_MSVC OR COMPILER_FAMILY_IS_MSVC_INTEL) )
       set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_MSVC} " )
-   elseif( DEFINED arg_DEFAULT )
-      set (${arg_FLAGS_VAR} "${${arg_FLAGS_VAR}} ${arg_DEFAULT} ")
-   endif()   
+   
+   #else, if we are setting a fortran flag, check against the fortran compiler family
```

Comment:
Am I understanding this properly:

If you want to get the flag for the fortran compiler, the letters ``fortran`` must be in the  ``FLAGS_VAR`` parameter's name (case insensitive)? 

If so, please document this, since it is a new behavior.

Reply:
Documented. Also protected the fortran stuff with ENABLE_FORTRAN so it falls back to the C compiler type regardless of flagvar name if fortran is disabled.